bin/magpie-hive-daemon.sh

```diff
@@ -0,0 +1,226 @@
+        if [ ! -d $HIVE_METASTORE_DIR ]
+        then
+            echo "Metastore warehouse directory ( ${HIVE_METASTORE_DIR} ) not found, creating directory now."
+            mkdir -p $HIVE_METASTORE_DIR
+
+            if [ ! -d $HIVE_METASTORE_DIR ] 
+            then
+                echo "Failed to create metastore warehouse directory at ${HIVE_METASTORE_DIR}! Unable to continue."
+                exit 1
+            fi
+        fi
+
+        echo "Starting Hive Metastore on ${myhostname}"
+
+        nohup $2/bin/hive --service metastore &> /dev/null &
+        scriptpid=$!
+        echo $scriptpid > ${HIVE_PID_DIR}/hive-metastore.pid
+        sleep 5
+        
+        export PATH=${PATH}:$2/bin
+        echo "Checking Hive Metastore settings"
+
+        # This will verify the NameNode location for the metastore, and if necessary update it to the correct node.
+        # EX: metastore was created/started on NODE-01 and was shut down, then brought back up on NODE-02.  metastore NameNode URI will contain hdfs://NODE-01/database 
+        # and NODE-02 will not connect properly.
+        # -listFSRoot finds and compares the NameNode in the metstore to the current node.  If necessary the -updateLocation function is called.
+        # $HIVE_HOME/bin/metatool -updateLocation <new location> <old location>  <-- this updates the metastore to the current NameNode
+        # EX: $HIVE_HOME/bin/metatool -updateLocation hdfs://n030:54310/path/to/metastore/database.db hdfs://n034:54310/path/to/metastore/database.db
+
+        fsroot=`${2}/bin/metatool -listFSRoot | grep 'hdfs://'`
+
+        for line in `echo $fsroot`; do
+            fshost=`echo "${line}"|cut -d / -f 3`
+            IFS=':' read -r -a tmparr <<<"$fshost"
+            name="${tmparr[0]}"
+            port="${tmparr[1]}"
+
+            if [[ "$name" != "$myhostname" ]]; then
+                hiveneedspathupdate=1
+                echo "Metastore reports ${fshost}, attempting to update to ${myhostname}"
+                newfshost="${line/$name/$myhostname}"
+                $2/bin/metatool -updateLocation ${newfshost} ${line}
+            else
+                echo "Metastore located at ${fshost}, update not required."
+                hiveneedspathupdate=0
+            fi
+        done
+    fi
+
+    metastorechk=`ps aux | grep '[H]iveMetaStore' | awk '{print $2}'`
+    if [ "${metastorechk}X" == "X" ]
+    then
+        echo "Hive Metastore failed to start! Unable to continue"
+        exit 1
+    fi
+
+    if [ -f "${HIVE_PID_DIR}/hive-server.pid" ]
+    then
+        echo "Hive server pid file found, it should already be running"
+    else
+        echo "Starting Hive server on ${myhostname}"
+
+        nohup $2/bin/hive --service hiveserver2 &> /dev/null &
+        scriptpid=$!
+        echo $scriptpid > ${HIVE_PID_DIR}/hive-server.pid
+        sleep 5
+        
+        # Check to see if the HIVE_DB_NAME database exists.  If not, create it.
```

Comment:
Normally this DB create would occur in ```magpie-setup-hive``` or ```magpie-run-project-hive```, but I assume it has to be done here b/c of the additional config down below in the script?